src/components/10-atoms/text/index.js

```diff
@@ -28,38 +31,51 @@ class AXAText extends NoShadowDOM {
     applyDefaults(this);
   }
 
-  render() {
-    // eslint-disable-next-line prefer-destructuring
-    const variant = this.variant;
-    const isSize2 = variant.indexOf('size-2');
-    const isSize3 = variant.indexOf('size-3');
-    const isBold = variant.indexOf('bold');
-
-    const classes = ['a-text'];
-    if (isSize2 === 0) {
-      classes.push('a-text--size-2');
-    }
-    if (isSize3 === 0) {
-      classes.push('a-text--size-3');
+  watch(mode) {
+    switch (mode) {
+      case 'stop':
+        if (this._observer) {
+          this._observer.disconnect();
+        }
+        break;
+      case 'start':
+        if (!this._observer) {
+          this._observer = new MutationObserver(() => this.render());
+        }
+        this._observer.observe(this, {
+          childList: true,
+        });
+        break;
+      default:
+        break;
     }
-    if (isBold === 0) {
-      classes.push('a-text--bold');
-    }
-
-    const { firstChild, firstElementChild } = this;
+  }
 
-    // nodeType === 3 is Text
-    if (firstChild && firstChild.nodeType === 3 && firstElementChild === null) {
-      const p = document.createElement('p');
-      p.innerHTML = firstChild.textContent;
-      p.className = classes.join(' ');
+  render() {
+    // do we have 'pure', non-empty text, e.g. <axa-text>Hello, World</axa-text>?
+    const nonWhitespaceTextNodes = [...this.childNodes].filter(
+      node => node instanceof Text && node.textContent.trim()
```

Comment:
@markus-walther better use NodeType here instead of relying on Class Type. 

https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

Reply:
Just be aware that u need to use the number, not the static property cause IE only knows the number type

Reply:
As discussed, will be changed in favour of nodeType.